src/lib.rs

```diff
@@ -342,6 +340,7 @@ impl IntoIterator for WalkDir {
 }
 
 /// A trait for recursive directory iterators.
+#[deprecated(note="This trait is no longer in use and and will be removed in the next major release. Its methods are now the inherent methods of `Iter` and `IterFilterEntry`")]
```

Comment:
I think we should add a `since` attribute here. #56 has just set that to `1.0.7`, so we could do that here.

From the [deprecation RFC](https://github.com/rust-lang/rfcs/blob/master/text/1270-deprecation.md) it looks like we maybe we should be setting this to `1.0.8`, but could do that before releasing a non-breaking `1.0.8` version.